### PR TITLE
Implements freezie floors and buffs frostoil

### PR DESCRIPTION
**Pull request by nullbear**
:cl: nullbear
rscadd: A reminder for crew to avoid mopping floors in cold rooms, as
the water will freeze and provide a dangerous slipping hazard known as
'ice'. Running on it is not recommended.
rscadd: Frostoil is an effective chemical for rapidly cooling areas
in the event of a fire.
/:cl:

Makes wet floors freeze in cold environments, and evaporate faster in warm
environments.

Makes frostoil cool the atmosphere, and make the floor wet. Typically
resulting in icy floors.

Fixes gluon grenade potentially making floors permanently icy.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
we use correct capitalization in changelogs pls

**Comment by nullbear** [comment]
wat.

**Comment by nullbear** [comment]
oops, didnt mean to hit that button

**Comment by Bawhoppen** [comment]
you just had to capitalize frostoil in the changelog
lowercase cls are shit

**Comment by nullbear** [comment]
capitalized frostoil looks funny

**Comment by Bawhoppen** [comment]
wait you messed with my gluon grenade? how long will it stay icy now? needs to stay icy for atleast 5 min

**Comment by nullbear** [comment]
Oop. I'll make it do that in a min

**Comment by nullbear** [comment]
Make it use permafrost instead of icy

**Comment by Bawhoppen** [comment]
also if you're in that file rename gluon grenade to 'gluon frag', meant to do that originally

**Comment by nullbear** [comment]
K, they will ice the area for 5-5.5 minutes. And then typically stay wet for a half minute or so after that as well.

**Comment by Bawhoppen** [comment]
wait are you able to see the ice? cause that is a big balance change for them
the idea was it was invisible ice and you slipped on it and then got killed from an op

**Comment by nullbear** [comment]
-sigh- 

I can change it so that your gluon grenades dont make visible ice.

**Comment by nullbear** [comment]
To recap, the gluon grenades will now ice the area for 5 minutes, invisibly. 

But ice is still visible, normally.

**Comment by Bawhoppen** [comment]
yes thanks perfecto

File: code/game/turfs/open.dm [hunk 1]
```diff
@@ -97,19 +97,39 @@
...
+	if(wet == TURF_WET_PERMAFROST)
+		wet = TURF_WET_ICE
+	else
+		wet = TURF_DRY
+		if(wet_overlay)
+			overlays -= wet_overlay
+
+/turf/open/proc/HandleWet()
+	if(!istype(src, /turf))
+		return
```
Comment (phil235) [inline code/game/turfs/open.dm]: whyyyyyyyyyyy

File: code/game/turfs/open.dm [hunk 2]
```diff
@@ -97,19 +97,39 @@
...
-		if(!istype(src, /turf)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
-			return
-		MakeDry(wet_setting)
+	if(wet < TURF_WET_ICE)
+		addtimer(src, "MakeDry", rand(790, 820))
 
 /turf/open/proc/MakeDry(wet_setting = TURF_WET_WATER)
-	if(wet > wet_setting)
+	if(!istype(src, /turf)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
+		return
```
Comment (phil235) [inline code/game/turfs/open.dm]: turfs don't become non turfs though...

Comment (nullbear) [inline code/game/turfs/open.dm]: I was thinking the same thing when i saw it. But i didnt add that, i just moved it.

**Comment by Shapsy** [comment]
Wait so this is like invisible lube?

**Comment by coiax** [comment]
No, it's visible ice, and modifies the gluon grenade to create invisible ice.

And I guess ice is equivalent to lube.

**Comment by Bawhoppen** [comment]
the gluon grenade already made permanent invisible ice so this just makes it temporary

**Comment by Zilenan91** [comment]
What the fuck why was the gluon grenade allowed in the game

**Comment by nullbear** [comment]
ice is like lube but much more high profile, and you can walk on it without slipping.

**Comment by Supermichael777** [comment]
no invisible slips. ignore bawling he's a hypocritical ass.

**Comment by nullbear** [comment]
k

**Comment by nullbear** [comment]
Done.

**Comment by Bawhoppen** [comment]
@Supermichael777 it was literally already in the game, this is just preserving the functionality
make a seperate PR later to nerf it but leave it as is right now @nullbear

**Comment by nullbear** [comment]
It's not anywhere as visible as ice. But still visible.

**Comment by Bawhoppen** [comment]
just change it back to being invisible and make a seperate balance PR

**Comment by Zilenan91** [comment]
No fuck off no invisible slips

**Comment by Bawhoppen** [comment]
zilenan they have been invisible since they were added which was quite awhile ago now, do not bundle a major rebalance with some good code changes
zilenan make your own seperate pr to change it but it shouldn't be bundled with a code improvement

**Comment by Zilenan91** [comment]
>No fuck off no invisible slips

**Comment by Bawhoppen** [comment]
@Zilenan91 oh hello mr.shitposts on PRs and never does anything and also doesn't actually play the game

**Comment by PKPenguin321** [comment]
funny that you'd say that baw

**Comment by Bawhoppen** [comment]
pkp I know you don't like me but I definitely do play and I don't shitpost as hard as zilenan

**Comment by nullbear** [comment]
if i just discard all changes to the file, it would only last about 2 minutes.

**Comment by Bawhoppen** [comment]
discard all other changes but let it last like 5 min atleast still sounds reasonable

**Comment by nullbear** [comment]
its literally just changing "wet = permafrost" to "makeslippery(permafrost)"

**Pull request merged**